server/services/ProfilesService.js

```diff
@@ -12,44 +12,56 @@ const uploadPhoto = function (photo, userId, dest) {
   })
 }
 
-const saveProfile = function (profile, userId) {
+const saveProfile = function(profile, userId) {
   const profilesRef = db.collection('users')
   const profileRef = db.collection('users').doc(userId)
+  const userPostsRef = db.collection('users-posts').doc(userId)
 
   return profileRef.get().then(doc => {
     const prevNickname = doc.data().nickname
     if (prevNickname === profile.nickname) {
-      return profileRef
-        .update({
-          bio: profile.bio,
-          status: profile.status,
-          interests: profile.interests,
-          avatar_url: profile.avatar_url,
-          socialAccounts: profile.socialAccounts,
-          user_background: profile.user_background
-        })
-        .then(() => {
-          return 'Profile edited successfully'
+      return db.runTransaction(t => {
+        return t.get(profileRef).then(profileDoc => {
+          return t.get(userPostsRef).then(userPostsDoc => {
+            t.update(profileRef, {
+              bio: profile.bio,
+              status: profile.status,
+              avatarUrl: profile.avatarUrl,
+              preferences: profile.preferences,
+              socialAccounts: profile.socialAccounts,
+              backgroundUrl: profile.backgroundUrl
+            })
+            t.update(userPostsRef, { avatarUrl: profile.avatarUrl })
+            return 'Profile saved successfully'
+          })
         })
+      })
     } else {
       return profilesRef
         .where('nickname', '==', profile.nickname)
         .get()
         .then(snapshot => {
           if (snapshot.empty) {
-            return profileRef
-              .update({
-                nickname: profile.nickname,
-                bio: profile.bio,
-                status: profile.status,
-                interests: profile.interests,
-                avatar_url: profile.avatar_url,
-                socialAccounts: profile.socialAccounts,
-                user_background: profile.user_background
-              })
-              .then(() => {
-                return 'Profile edited successfully'
+            return db.runTransaction(t => {
+              return t.get(profileRef).then(profileDoc => {
+                return t.get(userPostsRef).then(userPostsDoc => {
+                  t.update(profileRef, {
+                    nickname: profile.nickname,
+                    bio: profile.bio,
+                    status: profile.status,
+                    preferences: profile.preferences,
+                    avatarUrl: profile.avatarUrl,
+                    socialAccounts: profile.socialAccounts,
+                    backgroundUrl: profile.backgroundUrl
+                  })
+                  t.update(userPostsRef, {
```

Comment:
saveProfile shouldn't have update userPostsRef it should be a separate function which you could call on demand whenever avatar or username was changed.
Next, when user will change his profile he would actually wait until all his videos also updating, by saying this I mean is it nessecary to wait on client side until everything will be complete or simply only update profile, send a response and after that run separate function to update all videos.